File: core/MinimumSizeSubarraySum/MinimumSizeSubarraySum.py
from typing import List


def minSubArrayLen(target: int, nums: List[int]) -> int:
    left, right, length, total, res = 0, 0, len(nums), 0, 0
    while right < length:
        total += nums[right]
        while total >= target:
            res = right - left + 1 if res == 0 else min(res, right - left + 1)
            total -= nums[left]
            left += 1
        print(left, right)
        right += 1
    return res


if __name__ == '__main__':
    print(minSubArrayLen(11, [1,2,3,4,5]))

File: core/MinimumSizeSubarraySum/MinimumSizeSubarraySum.java
package com.dbc;

public class MinimumSizeSubarraySum {
    public int minSubArrayLen(int target, int[] nums) {
        int left = 0, right = 0, res = 0, total = 0;
        while (right < nums.length) {
            total += nums[right];
            while (total >= target) {
                res = res == 0 ? right - left + 1 : Math.min(res, right - left + 1);
                total -= nums[left];
                left++;
            }
            right++;
        }
        return res;
    }
}
